### ISSUE TITLE
[qgroundcontrol] Does not load (localization issues?)

### ISSUE THREAD

**Issue by DAMO238** [acxz/pkgbuilds#50]
Running qgroundcontrol in a terminal causes the following output and then goes unresponsive and cannot even be terminated with <C-c>:

```
warning: Environment variable QT_LOGGING_TO_CONSOLE is deprecated, use
QT_ASSUME_STDERR_HAS_CONSOLE and/or QT_FORCE_STDERR_LOGGING instead.
qt5ct: using qt5ct plugin

(QGroundControl:933475): Gtk-WARNING **: 17:26:55.256: Unable to locate theme engine in module_path: "adwaita",

(QGroundControl:933475): Gtk-WARNING **: 17:26:55.256: Unable to locate theme engine in module_path: "adwaita",

(QGroundControl:933475): Gtk-WARNING **: 17:26:55.257: Unable to locate theme engine in module_path: "murrine",

(QGroundControl:933475): Gtk-WARNING **: 17:26:55.257: Unable to locate theme engine in module_path: "murrine",

(QGroundControl:933475): Gtk-WARNING **: 17:26:55.257: Unable to locate theme engine in module_path: "murrine",

(QGroundControl:933475): Gtk-WARNING **: 17:26:55.258: Unable to locate theme engine in module_path: "murrine",

(QGroundControl:933475): Gtk-WARNING **: 17:26:55.258: Unable to locate theme engine in module_path: "murrine",

(QGroundControl:933475): Gtk-WARNING **: 17:26:55.258: Unable to locate theme engine in module_path: "murrine",

(QGroundControl:933475): Gtk-WARNING **: 17:26:55.258: Unable to locate theme engine in module_path: "murrine",

(QGroundControl:933475): Gtk-WARNING **: 17:26:55.258: Unable to locate theme engine in module_path: "murrine",

(QGroundControl:933475): Gtk-WARNING **: 17:26:55.258: Unable to locate theme engine in module_path: "murrine",

(QGroundControl:933475): Gtk-WARNING **: 17:26:55.258: Unable to locate theme engine in module_path: "murrine",

(QGroundControl:933475): Gtk-WARNING **: 17:26:55.258: Unable to locate theme engine in module_path: "murrine",
Settings location "/home/damien/.config/QGroundControl.org/QGroundControl.ini" Is writable?: true
Filter rules "*Log.debug=false\n"
MAVLinkLogManagerLog: MAVLink logs directory: "/home/damien/Documents/QGroundControl/Logs"
System reported locale: QLocale(English, Latin, United Kingdom) "en_GB"
Loading localization for "en_GB"
Qt localization for "en_GB" is not present
Map Cache in: "/home/damien/.cache/QGCMapCache300" / "qgcMapCache.db"
QQmlApplicationEngine failed to load component
qrc:/qml/MainRootWindow.qml:197:5: Type Drawer unavailable
file:///usr/lib/qt/qml/QtQuick/Controls.2/qmldir: plugin cannot be loaded for module "QtQuick.Controls": Cannot install type 'VerticalHeaderView' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'HorizontalHeaderView' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install element 'SplitHandle' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'SplitView' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install element 'Overlay' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'MenuBarItem' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'MenuBar' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ActionGroup' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Action' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ScrollView' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'DelayButton' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ToolSeparator' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'RoundButton' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'MenuSeparator' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'DialogButtonBox' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Dialog' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Tumbler' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ToolTip' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ToolButton' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ToolBar' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'TextField' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'TextArea' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'TabButton' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'TabBar' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'SwitchDelegate' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Switch' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'SwipeView' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'SwipeDelegate' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'StackView' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'SpinBox' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Slider' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ScrollIndicator' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ScrollBar' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'RangeSlider' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'RadioDelegate' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'RadioButton' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ProgressBar' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Popup' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Pane' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'PageIndicator' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Page' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'MenuItem' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Menu' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Label' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ItemDelegate' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'GroupBox' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Frame' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Drawer' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Dial' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Control' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Container' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ComboBox' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'CheckDelegate' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'CheckBox' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ButtonGroup' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Button' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'BusyIndicator' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ApplicationWindow' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'AbstractButton' into protected module 'QtQuick.Controls' version '2'
qt5ct: palette support is disabled
```

It appears to me that I need a certain qt module to give me localization, but I could not find such a module.

**Comment by acxz** [comment]
hmm interesting, I'm actually getting a different failure:
```
/opt/qgroundcontrol/QGroundControl: /opt/qgroundcontrol/Qt/libs/libQt5Core.so.5: version `Qt_5.15' not found (required by /usr/lib/libQt5WaylandClient.so.5)
/opt/qgroundcontrol/QGroundControl: /opt/qgroundcontrol/Qt/libs/libQt5Core.so.5: version `Qt_5.15' not found (required by /usr/lib/libQt5Concurrent.so.5)
/opt/qgroundcontrol/QGroundControl: /opt/qgroundcontrol/Qt/libs/libQt5Core.so.5: version `Qt_5.15' not found (required by /usr/lib/libQt5QmlModels.so.5)
```

**Comment by DAMO238** [comment]
Hmm, interesting, are fully up to date, qt 5.15 was released relatively recently?

**Comment by acxz** [comment]
Ah yep u right, just rebuilt it and got the following:
```
acxz@archard ~/.cache/yay/qgroundcontrol (git)-[master] % qgroundcontrol
warning: Environment variable QT_LOGGING_TO_CONSOLE is deprecated, use
QT_ASSUME_STDERR_HAS_CONSOLE and/or QT_FORCE_STDERR_LOGGING instead.
Settings location "/home/acxz/.config/QGroundControl.org/QGroundControl.ini" Is writable?: true
Filter rules "*Log.debug=false\n"
Error loading text-to-speech plug-in "speechd"
MAVLinkLogManagerLog: MAVLink logs directory: "/home/acxz/Documents/QGroundControl/Logs"
System reported locale: QLocale(English, Latin, United States) "en_US"
Loading localization for "en_US"
Qt localization for "en_US" is not present
Map Cache in: "/home/acxz/.cache/QGCMapCache300" / "qgcMapCache.db"
QQmlApplicationEngine failed to load component
qrc:/qml/MainRootWindow.qml:333:13: Type LogReplayStatusBar unavailable
qrc:/qml/QGroundControl/Controls/LogReplayStatusBar.qml:80:9: Type Slider unavailable
file:///usr/lib/qt/qml/QtQuick/Controls.2/qmldir: plugin cannot be loaded for module "QtQuick.Controls": Cannot install type 'VerticalHeaderView' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'HorizontalHeaderView' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install element 'SplitHandle' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'SplitView' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install element 'Overlay' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'MenuBarItem' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'MenuBar' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ActionGroup' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Action' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ScrollView' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'DelayButton' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ToolSeparator' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'RoundButton' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'MenuSeparator' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'DialogButtonBox' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Dialog' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Tumbler' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ToolTip' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ToolButton' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ToolBar' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'TextField' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'TextArea' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'TabButton' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'TabBar' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'SwitchDelegate' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Switch' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'SwipeView' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'SwipeDelegate' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'StackView' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'SpinBox' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Slider' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ScrollIndicator' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ScrollBar' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'RangeSlider' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'RadioDelegate' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'RadioButton' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ProgressBar' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Popup' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Pane' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'PageIndicator' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Page' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'MenuItem' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Menu' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Label' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ItemDelegate' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'GroupBox' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Frame' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Drawer' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Dial' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Control' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Container' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ComboBox' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'CheckDelegate' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'CheckBox' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ButtonGroup' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Button' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'BusyIndicator' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ApplicationWindow' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'AbstractButton' into protected module 'QtQuick.Controls' version '2'



```

**Comment by DAMO238** [comment]
Done: https://github.com/mavlink/qgroundcontrol/issues/8845

**Comment by acxz** [comment]
@DAMO238 Decided to get to the bottom of this and it was our issue all along.

Turns out the PKGBUILD was symlinking `qgroundcontrol-start.sh` to `qgroundcontrol` instead of the executable `QGroundControl`. `QGroundControl` works fine but `qgroundcontrol-start.sh` was causing the errors reported here.

I have fixed the PKGBUILD to symlink `QGroundControl` instead of `qgroundcontrol-start.sh` with `4.0.9-3`.
